### PR TITLE
Type annotation for util files

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: smac/utils/validate.py [hunk 1]
```diff
@@ -285,7 +298,7 @@ def validate_epm(self,
         elif reuse_epm is False or self.epm is None:
```
Comment (ashraaghav) [inline smac/utils/validate.py]: Might be nitpicky, but won't it be better to use `not reuse_epm` ?

[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/utils/validate.py]: Yep, changed that.

File: smac/utils/validate.py [hunk 2]
```diff
@@ -295,7 +295,7 @@ def validate_epm(self,
         """
         if not isinstance(runhistory, RunHistory) and (self.epm is None or reuse_epm is False):
```
Comment (ashraaghav) [inline smac/utils/validate.py]: same here too

File: smac/utils/validate.py [hunk 3]
```diff
@@ -282,10 +295,10 @@ def validate_epm(self,
...
```
Comment (ashraaghav) [inline smac/utils/validate.py]: same here too

**Pull request merged**